### PR TITLE
Mark current directory as safe

### LINKED ISSUE #49: Failed to read Git log: fatal: detected dubious ownership in repository at '/github/workspace'
Hello,

the action started to fail recently for one of our repos.
```
Building the Hugo site with: 'hugo '
Start building sites … 
hugo v0.98.0-...
ERROR 2022/12/20 10:14:35 Failed to read Git log: fatal: detected dubious ownership in repository at '/github/workspace'
To add an exception for this directory, call:
	git config --global --add safe.directory /github/workspace
Error: Error building site: logged 1 error(s)
```

I tried to reproduce it locally by checking out the code the way GitHub does and then running your action in a container as described in the README, but I haven't been successful so I don't have a clear reproducer for you.
But if you fork https://github.com/packit/status the https://github.com/packit/status/blob/main/.github/workflows/deploy-pages.yml should fail for you as well - if you change it to use your action instead of my fork and remove the `if:` condition.

I have been able to workaround the problem exactly the way the error message suggests, i.e. by [adding
"git config --global --add safe.directory ${PWD}" to the `action.sh`](https://github.com/jpopelka/hugo-deploy-gh-pages/commit/df7c2381c2e8837a0822007a08c808fd6661f152).

See actions/checkout#766 for more info.

### PR DESCRIPTION
**Pull request by jpopelka**
to work-around the following:
`ERROR Failed to read Git log: fatal: detected dubious ownership in repository at '/github/workspace'`

actions/runner#2033 describes the problem and actions/checkout#766 suggests to run
`git config --global --add safe.directory "$GITHUB_WORKSPACE"` inside the container action script.

The `$GITHUB_WORKFLOW` is, however, valid in the environment that runs our action in a container, but in the container it's the working directory, hence the `$PWD`.
For example in a workflow in my 'status' repo, the `$GITHUB_WORKSPACE` is `/home/runner/work/status/status` and the action is run in a container like this:
`docker run --workdir /github/workspace -v "/home/runner/work/status/status":"/github/workspace"`
i.e. `$GITHUB_WORKSPACE` (`/home/runner/work/status/status`) is mounted into `/github/workspace` in the container and the directory is set as working directory (`$PWD` inside the action script).

Fixes #49

### CODE REVIEW

File: action.sh [hunk 1]
```diff
@@ -77,6 +77,9 @@ if [[ -n "${GO_VERSION}" ]]; then
   go version
 fi
 
+# https://github.com/actions/checkout/issues/766
+git config --global --add safe.directory ${PWD}
```
[the remark this answers is not in the document]
Comment (jpopelka) [reply action.sh:81]: I've updated the commit comment. Hope it's better now.

[the remark this answers is not in the document]
Comment (benmatselby) [reply action.sh:81]: Hmm, it's still showing as `PWD`?

[the remark this answers is not in the document]
Comment (benmatselby) [reply action.sh:81]: Ah, understood, let me read it all, then we can get this merged.

**Review by benmatselby — APPROVED** [review]
Thanks for this contribution, appreciate it.

**Comment by benmatselby** [comment]
@jpopelka 

Would you mind fixing:

```
git config --global --add safe.directory ${PWD}
[7](https://github.com/benmatselby/hugo-deploy-gh-pages/actions/runs/3831173132/jobs/6520080430#step:3:8)
                                         ^----^ SC2086 (info): Double quote to prevent globbing and word splitting.
```

If not, I can get this sorted tomorrow.

**Comment by jpopelka** [comment]
fixed

**Pull request merged**